lib/active_record/immutable.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module Immutable
+    extend ActiveSupport::Concern
+
+    DEFAULT_PREVIOUS_VERSION_ID_COLUMN_NAME = 'previous_version_id'
+
+    included do
+      def self.default_scope
+        where(%{
+          NOT EXISTS (
+            SELECT 1
+            FROM #{quoted_table_name} model
+            WHERE model.#{DEFAULT_PREVIOUS_VERSION_ID_COLUMN_NAME} = #{quoted_table_name}.id
+          )
+        })
+      end
+
+      # active record override method for instead of updating, create a new version
```

Comment:
What do you think about rephrasing this to something like:
```
# Overrides ActiveRecord::Persistence#_update_record method
```

Reply:
much better